driver.go

```diff
@@ -63,15 +56,13 @@ type Output interface {
 	GetOutputPin(name string) (OutputPin, error)
 }
 
-//////////////////////////////////////////////////
-
-type PWMChannel interface {
+type Channel interface {
 	Name() string
 	Set(value float64) error
 }
 
 type PWM interface {
 	Driver
-	PWMChannels() []PWMChannel
-	GetPWMChannel(name string) (PWMChannel, error)
+	Channels() []Channel
```

Comment:
At a risk of running out of nouns, what do we want to call the pH or Analog... channels?

(Drivers may support multiple interfaces)

Reply:
Ack. I'll update.

Reply:
@theatrus i was hoping for something like this
```go
device.(hal.PWM).Channel("20").Set(1.2)
// and
device.(hal.ADC).Channel("12").Read()
```

or

```go
device.(hal.PWM).Set("20", 1.2)
// and
device.(hal.ADC).Read("12")
```
the thing I am trying to avoid is repeating PWM or ADC in multiple time. Albeit this is not a big deal, and may be I am just bike shedding,
```go
device.(hal.PWM).PWMChannel("20").Set( 1.2)
// and
device.(hal.ADC).ADCChannel("12").Read()
```

Reply:
Your API is much nicer, with the caveat that right now its allowable for a driver to both be PWM and ADC.

We can either disallow this (and register drivers per type?) or have to fallback to the prefix names.

Reply:
I dont see why we cant allow this. Consider this,
```go
type PWMChannel interface {
    Set(float64) error
}
type PWMDriver interface {
   Channel(string) (PWMChannel, error)
   Channels() []PWMChannel
}
type PWM interface {
  PWMDriver() PWMDriver
}

type ADCChannel interface {
    Read() (float64, error)
}
type ADCDriver interface {
   Channel(string) (ADCChannel, error)
   Channels() []ADCChannel
}
type ADC interface {
  ADCDriver() ADCDriver
}
type foo struct {}
(f *foo) ADCDriver()  {}...
(f *foo) PWMDriver() {} ...
```

Reply:
Yeah, we can return sub-drivers from the main driver, I'm ok with this.